### ISSUE TITLE
password input crashes lemurs

### ISSUE THREAD

**Issue by krysopath** [coastalwhite/lemurs#119]
When I supply my password to lemurs "password input box" the program crashes and discloses my password in clear text. 

![image](https://user-images.githubusercontent.com/6668411/215818002-90d5409d-0b22-4fbe-9f9a-763fc7ba7640.png)

my password is 38 bytes of printable ascii.

[CWE-209](https://cwe.mitre.org/data/definitions/209.html)

**Comment by coastalwhite** [comment]
Yup, that's a big issue. This was a TODO in the code that simply did not get done before. I make a PR to fix it. Let me clean up the code a bit tomorrow and I will merge it then.

**Comment by krysopath** [comment]
I compiled the linked branch and ran lemurs, I did not even try full password. just pressed `ö` on my pernidious `de` layout and it caused crash right after update.

looks like the branch does not fix the behaviour.

**Comment by coastalwhite** [comment]
I did unit-testing to verify the problem had been resolved, but that did not address the problem that happened while rendering. Anyway, I fixed the rendering issue now as well and cleaned the code up a bit.

The "ö" is not really ASCII and is instead a 2 byte UTF-8 character. Sadly, doing string manipulations on UTF-8 is a landmine for panics in Rust. If someone could improve this code, but currently I am kind of blanking. :sweat_smile:

Anyway, I really tried to make it stop `panic`ing.
